### DISCUSSION TITLE
Ros2 (docker) connection with Foxglove Studio

### DISCUSSION THREAD

**Discussion by RobinHeitz** [foxglove/community#623] [General]
Hi, 
I'm running Ros2 (humble) in a docker container (on my M1 Mac) and want to visualize it with foxglove.  I'm running the foxglove bridge and the output states:
`[INFO] [1675193346.087345799] [foxglove_bridge]: Starting foxglove_bridge with WebSocket++/0.8.2
[INFO] [1675193346.087462549] [foxglove_bridge]: [WS] Server running without TLS
[INFO] [1675193346.087777090] [foxglove_bridge]: [WS] WebSocket server listening at ws://0.0.0.0:8765
[INFO] [1675193346.087925174] [foxglove_bridge]: [WS] WebSocket server run loop started` - seems good to me.

But when I try to connect to ws://0.0.0.0:8765, I get "Connection failed" (both, in the app and chrome web app).
I run the docker container with `docker run -it --name foxglove --network=host foxglove-test`

My Dockerfile is the following:

```
FROM ros:humble
EXPOSE 8765
RUN sudo apt update && sudo apt install -y \
    ros-humble-foxglove-bridge \
    ros-humble-turtlesim

```
Any ideas? I'm really hoping that I can do ROS Stuff with my mac :(
I lost hope with RViz.

Thanks in advance, Robin.

**Comment by defunctzombie** [comment]
Unlike docker on linux, the `--network=host` does not do what you think it does. Docker VMs on mac run in a virtual machine. Instead of host networking you need to explicitly provide the ports to expose via `docker run -it --name foxglove -p 8765:8765 foxglove-test`.

I recommend reading up on other docker-on-mac gotchas on the docker docs and other online resources.

**Reply by RobinHeitz** [reply]
Thank you very much!
